modules/iam-primary-roles/main.tf

```diff
@@ -97,13 +105,102 @@ data "aws_iam_policy_document" "primary_roles_assume" {
       }
     }
   }
+
+  # Allow designated Spacelift roles to assume the Identity Ops role.
+  dynamic "statement" {
+    for_each = local.assume_role_restricted && each.key == "ops" ? ["spacelift"] : []
```

Comment:
This needs to be made optional. @osterman @aknysh Do we want to explicitly support Spacelift with a `spacelift_roles` input? If so, then make this statement conditional on 
```hcl
try(length(var.spacelift_roles), 0) > 0
```

Reply:
@Nuru we can generify it beyond spacelift since this doesn't need to be specific to them... Make this something like `ops_identity_assumable_roles` and it will be more easily seen as any external role that would need access to assume the ops identity role.